File: erpAdminSchool/src/Services/StudentServices.js
import http from '../Helpers/http'

export const Get = async () =>{
    const {data} = await http.get(`Students`);
    return data;
}

export const GetItem = async(Id) =>{
    const {data} = await http.get(`Students/${Id}`);
    return data;
}

export const Insert = async(Student) =>{
    const {data,status} = await http.post(`Students`,Student).catch(error=>{ return error.response });
    if( status === 400 ){
        data.errorNumber = status;
    }
    return data;
}

export const Update = async(Student) =>{
    const {data,status} = await http.put(`Students/${Student.studentId}`,Student).catch(error=>{ return error.response });
    if( status === 400 ){
        data.errorNumber = status;
    }
    return data;
}

export const Delete = async(Id) =>{
    const {data,status} = await http.delete(`Students/${Id}`).catch(error=>{ return error.response });
    if( status === 400 ){
        data.errorNumber = status;
    }
    return data;
}

File: ERPWebApi/Services/StudentServices.cs
using ERPWebApi.Data;
using ERPWebApi.Models;
using ERPWebApi.Services.Interfaces;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ERPWebApi.Services
{
    public class StudentServices : IBase<Student>
    {
        private ERPWebApiContext _eRPWebApi;

        public StudentServices(ERPWebApiContext eRPWebApi)
        {
            _eRPWebApi = eRPWebApi;
        }

      
        public async Task<IEnumerable<Student>> Get()
        {
          return  await _eRPWebApi.Student.Include(key => key.Scholarship).ToListAsync();
        }

        public async Task<Student> GetItem(int Id)
        {
            var student = await _eRPWebApi.Student.Include(key => key.Scholarship)
                                                  .SingleOrDefaultAsync(key => key.StudentId == Id);

            return student;
        }

        public async Task<Student> Insert(Student t)
        {
               _eRPWebApi.Student.Add(t);
            await _eRPWebApi.SaveChangesAsync();
            return t;
        }

        public async Task<Student> Update(int Id, Student t)
        {
              _eRPWebApi.ChangeTracker.Clear();
            _eRPWebApi.Entry(t).State = EntityState.Modified;
            await _eRPWebApi.SaveChangesAsync();
            return t;
        }

        public async Task Delete(int Id)
        {
            _eRPWebApi.ChangeTracker.Clear();
            _eRPWebApi.StudentSchoolSubject.RemoveRange(_eRPWebApi.StudentSchoolSubject.Where(key => key.StudentId == Id));
            _eRPWebApi.SaveChanges();
            _eRPWebApi.Student.Remove(new Student()
            {
                StudentId = Id
            });
             _eRPWebApi.SaveChanges();    
        }

    
    }
}
